setup/install_gatling/50_tune/10_puppet_infrastructure_tune.rb

```diff
@@ -1,25 +1,32 @@
 test_name 'Run puppet infrastructure tune' do
 
-  def run_puppet_infrastructure_tune
-    common_yaml = "/etc/puppetlabs/code-staging/environments/production/hieradata/common.yaml"
-
-    puts "Running 'puppet infrastructure tune' on master..."
+  def puppet_infrastructure_tune
+    base_tune_command = "puppet infrastructure tune"
+    common_yaml_path = "/etc/puppetlabs/code-staging/environments/production/hieradata/common.yaml"
+    tune_output_dir = "tune_output"
+    tune_output_path = "#{tune_output_dir}/nodes/#{master.hostname}.yaml"
+
+    # create tune_output dir
+    puts "Creating '#{tune_output_dir}' dir on master..."
     puts
+    on master, "mkdir -p #{tune_output_dir}"
 
-    output = on(master, "puppet infrastructure tune").output
-
-    # get data between '---' and the first empty line
-    data = output.match(/## Specify(.*)## CPU/m)[1].gsub("---", "").strip
-
-    # remove control codes
-    data = data.match(/0;32m(.*)\n.*\[/m)[1].strip + "\n"
+    # tune with hiera output and (and --force if specified)
+    hiera_option = " --hiera #{tune_output_dir}"
+    force_option = ENV["PUPPET_GATLING_SCALE_TUNE_FORCE"].eql?("true") ? " --force" : ""
+    tune_command = base_tune_command + hiera_option + force_option
+    puts "Running '#{tune_command}' on master..."
+    puts
+    on master, tune_command
+    tune_output = on(master, "cat #{tune_output_path}").output.gsub("---", "") + "\n"
 
-    puts "Extracted the following data:"
-    puts data
+    puts "Extracted the following output:"
+    puts tune_output
+    puts
 
     puts "Appending to common.yml"
     puts
-    on master, "echo \"#{data}\" >> #{common_yaml}"
+    on master, "echo \"#{tune_output}\" >> #{common_yaml_path}"
```

Comment:
If I read this right you are putting the tune results in both the common and the node specific yaml file?  If so why both?  Could make for confusion is someone tried to edit command and can't figure out why the edit doesn't work...

Reply:
It is using the '--hiera' option to generate the output file but it is just copying the contents of that file to the 'common.yaml' file. Getting the output from the file eliminates the extra output from the command. The 'tune_output' dir is just an arbitrary location to output the file.

Reply:
right, but isn't the file it is creating (the node file) enough to create the desired change?  So why also copy it to common?  Seems like the changes in common will have no effect.